docs/DeveloperGuide.md

```diff
@@ -1,34 +1,172 @@
+ AvailableModulesList availableModulesList) `
+`(AddToSemCommand#execute(SemesterList semesterList, AvailableModulesList availableModulesList))`
+
+##### Step 3:
+`AddToSemCommand#execute()` then calls self method `AddToSemCommand#addModule()`.`AddToSemCommand#addModule()`
+ then calls `AddToSemCommand#checkModuleExist(semesterList)` to check whether the selected 
+module is already in the selected module list (i.e:`semesterList`, which is a `PriorityQueue<SemModulesList>`). 
+If the module is not in the list, `AddToSemCommand#addModule()` will check whether there is a semester list
+(i.e:`semesterModulesList`, which is a `ArrayList<SelectedModule>`) whose name is the module's semester name. 
+If the semester list exist, the module will be added to the list. 
+If not, `AddToSemCommand#addModule()` will create a new semester list and then add this module to the new list. and the
+the new semester list will be added to `semesterList` as well.
+
+#### Step 4:
+`AddToSemCommand#execute()` calls `Ui.showAddedToSemMessage(selectedModule.announceAdded())` to show the result to the 
+user
+
+The following sequence diagram shows how the `Add to Semester` operation works:
+![Sequence Diagram of Add to Semester](https://github.com/RenzoTsai/tp/blob/Update_DG/docs/UML%20img%20folder/Sequence%20Diagram%20of%20Add%20to%20Semester.png)
+
+
+#### Calculate CAP feature
+
+The Calculate CAP mechanism is executed by `CalculateCapCommand`.  
+`CalculateCapCommand` is extended from `Command` and this implementation calculates the CAP using completed 
+`SelectedModule` stored in `SemModulesList`.
+
+Given below is the behaviour of the Calculate CAP mechanism at each step:
+
+Step 1: 
+User launches the application. `SelectedModules` are added to `SemModuleList` through either of the following methods:
+1) Imported from `semesterList.csv` using `StorageSemesterList.load()`
+2) Added using `add id/ID s/SEMESTER mc/MODULE_CREDIT` command
+
+Step 2:
+User executes `CAP` command to view his own CAP. The `CAP` commands is parsed through `Parser`, which would then return 
+`CalculateCapCommand()`. `CalculateCapCommand.execute()` is then called.
+
+Step 3:
+`CalculateCapCommand.execute()` will call `CalculateCapCommand.calculateCap(SemesterList semesterList)`, which will
+calculate CAP by looking for all the completed `SelectedModules` stored within `SemModuleList`, which are stored within 
+`SemesterList`. It will then assign a `double` type ranging from 0.00 to 5.00 to `Person.totalCap`.
+
+Step 4:
+After the CAP is assigned to `Person.totalCap`, `Person.totalCap` is then called and formatted using `DecimalFormat` into a `String`
+with a pattern of `#.00`. `Ui.showcap(cap)` is called to display the user's cap using the formatted `String`.
+
+The following diagram shows how the Calculate CAP operation works:
+![Calculate CAP feature](https://github.com/bennychanya/tp/blob/master/CalculateCap.png?raw=true)
```

Comment:
I agree with what Isabella said about the construction of CalculateCapCommand. Otherwise it looks good to me. Perhaps you may want to specify the return value (grade) of the getGrade() method to be consistent, as you did so for totalCap?